### ISSUE TITLE
review of GloBI's Darwin Core-ish data product

### ISSUE THREAD

**Issue by jhpoelen** [globalbioticinteractions/globalbioticinteractions#729]
@tucotuco shared some personal notes/ questions about the GloBI's Darwin Core-ish data product listed on https://globalbioticinteractions.org/data . 

Hoping to move this discussion to this public forum to that others can listen in and contribute (if desired).

**Comment by jhpoelen** [comment]
Here's some of the comments that @tucotuco shared in an email dated 2021-11-13 - 

"I have since looked at the Darwin Core Archive version of GloBI you publish at https://zenodo.org/record/4460654/files/dwca.zip in an attempt to understand if the approach to interactions I have taken would support what you and others might publish. That archive isn't a valid Darwin Core Archive for many reasons, so I am curious why you publish it. In any case, from it I think I can surmise the intended data publishing model at least. I believe it to be the following, as expressed in an ER diagram, with liberty taken to leave in only the things I need to remember, plus the liberty to abbreviate a sets of terms with their class as a placeholder:



I am also curious why you opted for a new Association class when there was already a ResourceRelationship to do what the Association does. But most of all I just want to know if this publishing model matches your intent correctly since the Darwin Core Archive can't actually express all of that. 

![image](https://user-images.githubusercontent.com/1084872/142087593-915d4d0e-04bb-4d31-ab3d-b1a16ec98bea.png)


If we weren't constrained by Occurrences (as in, they ceased to exist as a class), could you see Associations being an Event? Or an assertion about an Event? The difference is intriguing, and though it actually looks the same either way in the model I have created, there are distinct implications in terms of content. Here is a draft of the model as it stands right now.

![image](https://user-images.githubusercontent.com/1084872/142087581-29a94b1b-9b27-4363-8ee2-aef15cc32695.png)



and a simplified version of it for publishing purposes:

![image](https://user-images.githubusercontent.com/1084872/142087571-44b183c0-d10f-4a20-9540-7ecff03555ab.png)

I'm curious about any thoughts you wish to share."

**Comment by jhpoelen** [comment]
and on 2021-11-14 @tucotuco shared by email: 

"I have since found the answer to the question of why publish the invalid Darwin Core Archive - EOL."

**Comment by jhpoelen** [comment]
and the context is:

"to come up with a model by which the limitations of the Darwin Core Archive could be overcome and people could publish data that are currently a challenge to share without loss of information. Interactions is just one use case - the twelfth so far - among what will likely be more than twenty. There are several videos of presentations Tim has given describing the project. The first (https://vimeo.com/564600741) was a preview from before the project began, given at the most recent GBIF Global Nodes Meeting in June. He also presented on progress (https://gbif.app.box.com/s/7cr98uozrgrs88lz2r0lgt8w5z4yanyw) to the GBIF Governing Board in October."

**Comment by jhpoelen** [comment]
related to #730

**Comment by jhpoelen** [comment]
@tucotuco for a historic perspective on the development of the DarwinCore-ish archive that GloBI publishes, please see :

https://github.com/globalbioticinteractions/globalbioticinteractions/issues/16

https://github.com/globalbioticinteractions/globalbioticinteractions/issues/79

https://github.com/globalbioticinteractions/globalbioticinteractions/issues/165

**Comment by jhpoelen** [comment]
@tucotuco asks:

> If we weren't constrained by Occurrences (as in, they ceased to exist as a class), could you see Associations being an Event? Or an assertion about an Event?

I see species interaction records as description of interaction events. Also, I see interaction records as  interpretations of existing descriptions of interaction events. Many interaction data have rich (and often inconsistent) layers of interpretations.   

So, to answer your question, yes, I can see associations being an events. And, yes, I can also see some associations being assertions of such events.

**Comment by jhpoelen** [comment]
@tucotuco am open to your suggestions to improve the DwC-ish product. So far, EOL is able to use it, so I haven't had a reason to change it.

**Comment by tucotuco** [comment]
As I understand the history of it, the archive was made to satisfy sharing with EOL and it has served that purpose. My comment was based on the premise that it was a Darwin Core Archive as labeled. But it isn't valid, and can't be if it has no core element and violates the star schema constraint. Given its purpose and the fact that it works for that purpose, I wouldn't change anything except the labelling that represents it as a Darwin Core Archive.

Not sure if you want to coninue discussion of other aspects of the email exchange here or in some other issue.

**Comment by jhpoelen** [comment]
hey @tucotuco - yes, I appreciate your comment that the dwca should more be labeled as dwca-ish, or resembling a dwca. 

And, I would appreciate specific feedback items (with examples), just to tap into your extensive knowledge about the dwca. I might just find some time to bring the file "up to code" with your detailed comments without breaking the live integration with EOL.

**Comment by tucotuco** [comment]
Hi @jhpoelen,

The meta.xml file declares that it is following a schema (http://services.eol.org/schema/dwca/tdwg_dwc_text.xsd) that is not the Darwin Core text schema (http://rs.tdwg.org/dwc/text/tdwg_dwc_text.xsd). The EOL one is not responding, so I can't tell if the meta.xml file complies with it. Assuming that it does, there are differences between it and the Darwin Core text schema. First, the Darwin Core text schema requires that a core element be declared:
```
<xs:element name="core" type="arch:coreFileType" maxOccurs="1" minOccurs="1"/>
```
https://github.com/tdwg/dwc/blob/c545c438c89ccd21c6c26668dc61dc08a6288ed4/docs/text/tdwg_dwc_text.xsd#L20

The meta.xml file in your zip files do not contain a `core` element. Nor do they contain any `extension` elements, which, though they are not required, are the way to connect files to the one designated by the `core` element.

```
<xs:element name="extension" type="arch:extensionFileType" maxOccurs="unbounded" minOccurs="0"/>
```
https://github.com/tdwg/dwc/blob/c545c438c89ccd21c6c26668dc61dc08a6288ed4/docs/text/tdwg_dwc_text.xsd#L21

In place of both `core` and `extension` elements, your meta.xml files has `table` elements, which do not exist in the Darwin Core as Text specification, but clearly fulfill a similar role.

As far as I can tell, your meta.xml file has no way to declare what the relationships should be between the files as would be the case with the `core` and `extension` elements. That must be done on the EOL side by some kind of convention that is understood and instantiated in the ingest implementation.

The bottom line is that I don't think you can do anything to those files without breaking something with EOL, and that I can't tell much more without being able to see the XML Schema the meta.xml files declares in its schemaLocation.

**Comment by jhpoelen** [comment]
@tucotuco Thanks for sharing your detailed review notes. Perhaps a novel way to encode/package this corpus of indexed species interaction data can serve as a testing ground for whatever scheme you come up with. The diverse kinds of data records and origins, as well as the nimble development researchers would represent a realistic litmus test for such a scheme.  I am up to try out pragmatic ways to exchange species interaction data, just like what happened with EOL back in 2014.
